agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,228 @@
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            if (csvBeanReader != null) {
+                try {
+                    csvBeanReader.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+        }
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+
+        String requestCompanyName = request.getText().toLowerCase().replace("stocks ", "").replace("stock ", "");
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
+        String keywordCompanyName = stockCompanyInfo[0];
+        String keywordCompanySymbol = stockCompanyInfo[1];
+        String json = null;
+
+        if (keywordCompanySymbol == null) {
+            return new AgentResponse("I am sorry, I don't understand which company you are asking about.");
+        } else {
+            json = requestData(keywordCompanySymbol);
+        }
+
+        ArrayList<JsonElement> responseArray = new ArrayList<>();
+        ArrayList<JsonElement> responseLabelsArray = new ArrayList<>();
+
+        Gson gson = new Gson();
+
+        try {
+            JsonObject jsonObj = gson.fromJson(json.toString(),
+                    JsonObject.class);
+
+            JsonArray jsonDataArray = (JsonArray) (((JsonObject) jsonObj
+                    .get("dataset")).get("data"));
+
+            for (int i = 0; i < jsonDataArray.size(); i++) {
+                JsonArray jsonDataArrayNested = (JsonArray) (jsonDataArray
+                        .get(i));
+                responseArray.add(jsonDataArrayNested.get(4));
+                responseLabelsArray.add(jsonDataArrayNested.get(0));
+            }
+            Collections.reverse(responseArray);
+            Collections.reverse(responseLabelsArray);
+
+            List<JsonElement> returnDataArray = (responseArray.subList(
+                    responseArray.size() - 30, responseArray.size()));
+            List<JsonElement> returnLabelsArray = (responseLabelsArray
+                    .subList(responseLabelsArray.size() - 30,
+                            responseLabelsArray.size()));
+
+            JsonObject returnObj = gson.fromJson(json.toString(),
+                    JsonObject.class);
+
+            returnObj.addProperty("plotType", "lineplot");
+            returnObj.addProperty("data", returnDataArray.toString());
+            returnObj.addProperty("labels", returnLabelsArray.toString());
+            returnObj.addProperty("title", keywordCompanyName + " Stock Ticker");
+            returnObj.addProperty("xaxis", "Day");
+            returnObj.addProperty("yaxis", "Close of day market price ($)");
```

Comment:
Fixed. Added Json creation code to processData(String json, String keywordCompanyName).